dbt/adapters/bigquery.py

```
@@ -0,0 +1,300 @@
+
+        query.run()
+
+        return cls.fetch_query_results(query)
+
+    @classmethod
+    def add_begin_query(cls, profile, name):
+        raise Exception("not implemented")
+        return cls.add_query(profile, 'BEGIN', name, auto_begin=False,
+                             select_schema=False)
+
+    @classmethod
+    def create_schema(cls, profile, schema, model_name=None):
+        logger.debug('Creating schema "%s".', schema)
+
+        dataset = cls.get_dataset(profile, schema, model_name)
+
+        with cls.exception_handler(profile, 'create dataset', model_name):
+            dataset.create()
+
+    @classmethod
+    def check_schema_exists(cls, profile, schema, model_name=None):
+        conn = cls.get_connection(profile, model_name)
+
+        client = conn.get('handle')
+
+        with cls.exception_handler(profile, 'create dataset', model_name):
+            all_datasets = client.list_datasets()
+            return any([ds.name == schema for ds in all_datasets])
+
+    @classmethod
+    def get_dataset(cls, profile, dataset_name, model_name=None):
+        conn = cls.get_connection(profile, model_name)
+
+        client = conn.get('handle')
+        dataset = client.dataset(dataset_name)
+        return dataset
+
+    @classmethod
+    def add_query(cls, profile, sql, model_name=None, auto_begin=True):
+        raise Exception("Not implemented")
+
+    @classmethod
+    def cancel_connection(cls, profile, connection):
+        raise Exception("Not implemented")
+        handle = connection['handle']
+        sid = handle.session_id
+
+        connection_name = connection.get('name')
+
+        sql = 'select system$abort_session({})'.format(sid)
+
+        logger.debug("Cancelling query '{}' ({})".format(connection_name, sid))
+
+        _, cursor = cls.add_query(profile, sql, 'master')
+        res = cursor.fetchone()
+
+        logger.debug("Cancel query '{}': {}".format(connection_name, res))
```

what's the behavior on ctrl+c here?

if we need to just raise an exception, you should remove all this code.

also please change this and the exception for `add_query` to dbt.exceptions.NotImplementedException